src/Control/Monad/Aff/Bus.purs

```diff
@@ -54,18 +57,16 @@ type BusW' r = Bus (write ∷ Cap | r)
 type BusRW = Bus (read ∷ Cap, write ∷ Cap)
 
 -- | Creates a new bidirectional Bus which can be read from and written to.
-make ∷ ∀ eff a. Aff (avar ∷ AVAR | eff) (BusRW a)
-make = do
-  cell ← makeEmptyVar
-  consumers ← makeVar mempty
-  let
-    loop = do
-      attempt (takeVar cell) >>= traverse_ \res → do
-        vars ← takeVar consumers
-        putVar mempty consumers
-        sequence_ (foldl (\xs a → putVar res a : xs) mempty vars)
-        loop
-  _ ← forkAff loop
+make ∷ ∀ m eff a. MonadEff (avar ∷ AVAR | eff) m => m (BusRW a)
+make = liftEff do
+  cell ← EffAvar.makeEmptyVar
+  consumers ← EffAvar.makeVar mempty
+  runAff_ (const $ pure unit) $ fix \loop →
```

Comment:
I don't know if this use of `fix` is stack-safe like the Aff version?

Reply:
Oh yeah I see. Well it did confuse me for one :D But doesn't this still introduce extra thunking because of Lazy's `defer` at every level?